vendor/github.com/containers/image/docker/docker_client.go

```diff
@@ -146,21 +156,22 @@ func (c *dockerClient) setupRequestAuth(req *http.Request) error {
 			// no need for bearer? wtf?
 			return nil
 		}
-		tokens = strings.Split(hdr, " ")
-		tokens = strings.Split(tokens[1], ",")
-		var realm, service, scope string
-		for _, token := range tokens {
-			if strings.HasPrefix(token, "realm") {
-				realm = strings.Trim(token[len("realm="):], "\"")
-			}
-			if strings.HasPrefix(token, "service") {
-				service = strings.Trim(token[len("service="):], "\"")
-			}
-			if strings.HasPrefix(token, "scope") {
-				scope = strings.Trim(token[len("scope="):], "\"")
-			}
+		re := regexp.MustCompile(`^Bearer realm="(.*)",service="(.*)",scope="(.*)"$`)
```

Comment:
FWIW this probably won’t work, with OpenShift I can now see
`Bearer realm="http://127.0.0.1:5000/openshift/token"` (i.e. no service, no scope)